bindings/pyroot/pythonizations/python/ROOT/_pythonization/_function_utils.py

```diff
@@ -0,0 +1,224 @@
+        elif isinstance(x, float):
+            return 'float'
+        elif isinstance(x, np.ndarray): 
+            if x.dtype.type in  map_numpy_to_tree:
+                return "RVec<" + map_numpy_to_tree[x.dtype.type] + ">"
+            else:
+                raise TypeError(f"Support for {x.dtype.type} arrays is not yet supported.")
+        #! Need to work out how to map things like tuples, dicts, lists...
+        else:
+                raise TypeError(f"Type of {type(x)}:  {x} is not supported to be jitted.")
+            
+    def find_function_signature(self, func, funcargs):
+        """
+        Function to determine the function signature i.e the type of variables the function arguments and its corresponding value.
+        Args:
+	    `   func: Function whose signature is to be determined.
+                    P.S The function should not be crazy. It should have simple datatypes for the arguments and should not have default arguments. (Use funcargs instead)
+            funcargs: Dictionatry containing constant parameters the that func requires. Or any special mapping of the arguments of func.
+
+        Returns:
+            input_args: Map of func's input variables to values
+            type_args : Map of func'c input variables to its corresponding type.
+        """
+        try:
+            import inspect 
+        except:
+            raise ImportError("Failed to import inspect during call to determine function signature.")
+        func_sign = inspect.signature(func)
+        # Find the Return type
+        if func_sign.return_annotation is inspect._empty:
+            raise ValueError("Return type of the function is not mentioned.\n Function cannot be jitted as signature cannot be determined.")
+        else:
+            self.return_type = str(func_sign.return_annotation)
+        
+        # List of input parameters for function
+        self.params = list(func_sign.parameters.keys()) # ALl the input parameters the function needs.
+        input_args = {}
+        type_args = {}
+
+        # if len(funcargs) != 0:
+        for p in self.params:
+            if p in funcargs.keys(): # the parameter value has been given in funcargs
+                p_value = funcargs[p]
+                type_args[p] = self.find_type(p_value)
+                if type(p_value) == bool: # Bools in python are represented as True/False but in C++ are true/false. The followinf if is to account for that
+                    if p_value: p_value = 'true'
+                    else: p_value = 'false'
+                input_args[p] = p_value
+            else: # the parameter was not in funcargs. Thus this parameter has to be mapped to a column of rdf
+                if p not in self.col_names:
+                    raise Exception(f"Unable to map function argument {p} to a column.\nUse correct name of column or use funcargs.")
+                input_args[p] = p 
+                type_args[p] = self.find_type(p)
+        return input_args, type_args
```

Comment:
This part should be reorganized imho, possibly using only one dictionary instead of two. The dictionary should look like
```python
args_info : Dict[str, Tuple[str, str]] = { PARAM: (VALUE, TYPE) }
```